src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -4973,6 +4981,20 @@ public Expression GenerateCallContains(Expression lhs, Expression rhs, bool igno
                 rhs.Cast(typeof(object)));
         }
 
+        public object VisitTernaryExpression(TernaryExpressionAst ternaryExpressionAst)
+        {
+            var expr = Expression.Condition(
+                CaptureAstResults(
+                    CallAddPipe(
+                        Compile(ternaryExpressionAst.Condition),
+                        s_getCurrentPipe),
+                    CaptureAstContext.Condition).Convert(typeof(bool)),
+                Compile(ternaryExpressionAst.IfTrue).Convert(typeof(object)),
```

Comment:
What's the purpose of `Convert(typeof(object))` here?

Reply:
`Expression.Condition` expects both branches return the same type, so we need to cast to object here.